server/src/main/java/io/crate/statistics/Samples.java

```diff
@@ -26,129 +26,87 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Random;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
 
-import io.crate.Streamer;
-import io.crate.common.collections.Lists;
-import io.crate.data.Row;
-import io.crate.data.RowN;
 import io.crate.metadata.ColumnIdent;
 import io.crate.metadata.Reference;
-import io.crate.types.DataType;
 
 class Samples implements Writeable {
 
-    static final Samples EMPTY = new Samples(List.of(), List.of(), 0L, 0L);
+    static final Samples EMPTY = new Samples(List.of(), 0L, 0L);
 
-    @SuppressWarnings("rawtypes")
-    private final List<Streamer> recordStreamer;
-    private final List<Row> records;
+    private final List<ColumnSketchBuilder<?>> columnSketches;
     private final long numTotalDocs;
     private final long numTotalSizeInBytes;
 
-    @SuppressWarnings("rawtypes")
-    Samples(List<Row> records, List<Streamer> recordStreamer, long numTotalDocs, long numTotalSizeInBytes) {
-        this.records = records;
-        this.recordStreamer = recordStreamer;
+    Samples(List<ColumnSketchBuilder<?>> columnSketches, long numTotalDocs, long numTotalSizeInBytes) {
+        this.columnSketches = columnSketches;
         this.numTotalDocs = numTotalDocs;
         this.numTotalSizeInBytes = numTotalSizeInBytes;
     }
 
-    @SuppressWarnings("rawtypes")
-    public Samples(List<Streamer> recordStreamer, StreamInput in) throws IOException {
-        this.recordStreamer = recordStreamer;
+    public Samples(List<Reference> references, StreamInput in) throws IOException {
+        if (in.getVersion().before(Version.V_5_7_0)) {
+            throw new UnsupportedOperationException("Cannot run ANALYZE in a mixed version cluster");
```

Comment:
Does this need some special handling in `TableStatsService` ?

Reply:
Oh that's a good idea, we can add a line to `update` that returns early if the minimum node version in the cluster is before 5.7.  I'll push an update.